x-pack/plugins/canvas/public/state/middleware/aeroelastic.js

```diff
@@ -320,6 +322,14 @@ export const aeroelastic = ({ dispatch, getState }) => {
       aero.removeStore(action.payload);
     }
 
+    if (pageChangerActions.indexOf(action.type) >= 0) {
+      if (getSelectedElement(getState())) {
```

Comment:
Isn't this the same as `prevPage` from above (which you're also using in the else condition)?

Reply:
`prevPage` is bound to the result of `getSelectedPage`, not `getSelectedElement` to which the comment seems attached. 

In any case, the overall point about common subexpression elimination is true. I had a provisional commit, which extracts out all the `getState()` etc. calls in this file, in some cases, reused like 10 times. I ended up not pushing it because this middleware file is mostly going away if the local (`workpadPage`) state based approach proves solid, and there is no notable perf difference. 

Also, my mental model is slightly hazy if there are middleware statements that dispatch Redux actions: the dispatched action will re-execute the middleware part (I believe synchronously), and coming _back_ from that dispatch to the original middleware call, what you have in `getState` and what you might have _originally_ bound `getState` to might deviate. Ideally it's not the case (ie. calling `getState` from within the same execution of the middleware function should yield the snapshot) but again, I'm lacking a clear model, and I don't currently see how either way can be semantically clear (`getState()` giving old results, or giving new results both have their problems), so I'd be glad to hear or read info on this.